queries.sql

```diff
@@ -21,3 +21,102 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+
+-- Operation 1
+BEGIN;
+-- Update species to 'unspecified'
+UPDATE animals
+SET species = 'unspecified';
+-- Verify the change
+SELECT * FROM animals;
+-- Rollback the change
+ROLLBACK;
+-- Verify species column reverted to previous state
+SELECT * FROM animals;
+
+
+
+-- Operation 2
+BEGIN;
+-- Update species to 'digimon' for names ending in 'mon'
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+-- Update species to 'pokemon' for animals without a species
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+-- Verify the changes
+SELECT * FROM animals;
+-- Commit the transaction
+COMMIT;
+-- Verify changes persist after commit
+SELECT * FROM animals;
+
+
+
+-- Operation 3
+-- Start a transaction
+BEGIN;
+
+-- Delete all records in the animals table
+DELETE FROM animals;
+
+-- Rollback the transaction
+ROLLBACK;
+
+-- Check if records still exist after rollback
+SELECT * FROM animals;
+
+
+
+-- Operation 4
+BEGIN;
+-- Delete animals born after Jan 1st, 2022
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+-- Create a savepoint
+SAVEPOINT my_savepoint;
+-- Update weights to be weight * -1
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+-- Rollback to the savepoint
+ROLLBACK TO my_savepoint;
+-- Update negative weights to be weight * -1
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+-- Commit the transaction
+COMMIT;
+-- Verify changes persist after commit
+SELECT * FROM animals;
+
+
+
+-- Operation 5
+-- - How many animals are there?
+SELECT COUNT(*) FROM animals;
+
+-- - How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+-- - What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- - Who escapes the most, neutered or not neutered animals?
+SELECT neutered, SUM(escape_attempts) AS total_escape_attempts
+FROM animals
+GROUP BY neutered;
```

Comment:
- [ ] [OPTIONAL]
Let's bear in mind that the `SUM` function will return total sum of escape attempts for each "neutered" status instead of directly returning the results of which animal group escaped the most. It will be more appropriate to use query similar to the below:

```sql
  SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```

or

```sql
  SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
```

Reply:
Noted